### PR TITLE
wrapped Invoke-Sql calls in try/catch blocks and re-throw exception

### PR DESCRIPTION
**Pull request by bsmithvt**

### CODE REVIEW

File: Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1 [hunk 1]
```diff
@@ -303,8 +305,14 @@ function Get-IdentityDatabaseConnectionString([string] $identityDbName, [string]
...
     $identityDbConnStr = "Server=$($sqlServerAddress);Database=$($identityDbName);Trusted_Connection=True;MultipleActiveResultSets=True;"
 
-    Invoke-Sql $identityDbConnStr "SELECT TOP 1 ClientId FROM Clients" | Out-Null
-    Write-DosMessage -Level "Information" -Message "Identity DB Connection string: $identityDbConnStr verified"
+    try {
+        Invoke-Sql $identityDbConnStr "SELECT TOP 1 ClientId FROM Clients" | Out-Null
+        Write-DosMessage -Level "Information" -Message "Identity DB Connection string: $identityDbConnStr verified"
+    }
+    catch {
+        Write-Error $_.Exception -ErrorAction Stop
```
Comment (michaelvidal24) [inline Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: Why can't we use Write-DosMessage -Level "Fatal" here?

File: Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1 [hunk 2]
```diff
@@ -318,8 +326,14 @@ function Get-MetadataDatabaseConnectionString([string] $metadataDbName, [string]
...
     $metadataConnStr = "Server=$($sqlServerAddress);Database=$($metadataDbName);Trusted_Connection=True;MultipleActiveResultSets=True;"
 
-    Invoke-Sql $metadataConnStr "SELECT TOP 1 RoleID FROM CatalystAdmin.RoleBASE" | Out-Null
-    Write-DosMessage -Level "Information" -Message "Metadata DB Connection string: $metadataConnStr verified"
+    try {
+        Invoke-Sql $metadataConnStr "SELECT TOP 1 RoleID FROM CatalystAdmin.RoleBASE" | Out-Null
+        Write-DosMessage -Level "Information" -Message "Metadata DB Connection string: $metadataConnStr verified"
+    }
+    catch {
+        Write-Error $_.Exception -ErrorAction Stop
```
Comment (michaelvidal24) [inline Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: Same comment as above...

File: Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1 [hunk 3]
```diff
@@ -26,6 +26,8 @@ else {
     Import-Module -Name $dosInstallUtilities.FullName
 }
 
+$ErrorActionPreference = "Stop"
```
Comment (michaelvidal24) [inline Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: By setting this, the default action on any error is going to be stop...which I think is largely fine...

File: Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1 [hunk 4]
```diff
@@ -548,7 +562,13 @@ function Add-DatabaseLogin([string] $userName, [string] $connString)
                 set @sql = 'CREATE LOGIN ' + QUOTENAME('$userName') + ' FROM WINDOWS'
                 EXEC sp_executesql @sql
             END"
-    Invoke-Sql $connString $query @{userName=$userName} | Out-Null
+    try {
+        Invoke-Sql $connString $query @{userName=$userName} | Out-Null
+    }
+    catch {
+        Write-Error $_.Exception -ErrorAction Continue
```
Comment (michaelvidal24) [inline Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: Is there a reason we are stopping for some and continuing for other?

Comment (bsmithvt) [reply Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: The original `Install-Identity-Windows` script had try/catch blocks for some `Invoke-Sql` calls while omitted it on others. Do we always want to halt if we can't do any SQL-related activity, including, for example, migrating data?

File: Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1 (hunk 2 above)
Comment (bsmithvt) [reply Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: We can change this. We are using `Write-Error` in some places. I'll change to use the Dos library function.

Comment (michaelvidal24) [reply Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1]: We use Write-Error in Fabric-Install-Utilities, we shouldn't be using it anywhere else, we want to move to DosInstallUtilities for new stuff.

**Pull request merged**